examples/step-69/step-69.cc

```diff
@@ -0,0 +1,355 @@
+
+  static inline DEAL_II_ALWAYS_INLINE double pressure(const rank1_type U);
+
+  static inline DEAL_II_ALWAYS_INLINE double
+  specific_entropy(const rank1_type U);
+
+  static inline DEAL_II_ALWAYS_INLINE double entropy(const rank1_type U);
+
+  static inline DEAL_II_ALWAYS_INLINE rank1_type
+                                      entropy_derivative(const rank1_type U);
+
+  static inline DEAL_II_ALWAYS_INLINE rank2_type f(const rank1_type U);
+};
+
+
+// FIXME Unify
+template <int dim>
+class RiemannSolver
+{
+public:
+  static constexpr unsigned int problem_dimension =
+    ProblemDescription<dim>::problem_dimension;
+
+  using rank1_type = typename ProblemDescription<dim>::rank1_type;
+
+  RiemannSolver(const std::string &subsection = "RiemannSolver");
+
+  static std::
+    tuple<double /*lambda_max*/, double /*p_star*/, unsigned int /*iteration*/>
+    compute(const rank1_type              U_i,
+            const rank1_type              U_j,
+            const dealii::Tensor<1, dim> &n_ij);
+};
+
+
+template <int dim>
+class InitialValues : public dealii::ParameterAcceptor
+{
+public:
+  static constexpr unsigned int problem_dimension =
+    ProblemDescription<dim>::problem_dimension;
+
+  static constexpr double gamma = ProblemDescription<dim>::gamma;
+
+  static constexpr double b = ProblemDescription<dim>::b;
+
+  typedef dealii::Tensor<1, problem_dimension> rank1_type;
+
+  InitialValues(const std::string &subsection = "InitialValues");
+
+  void parse_parameters_callback();
+
+  const std::function<rank1_type(const dealii::Point<dim> &point, double t)>
+    &initial_state;
+
+
+private:
+  dealii::Tensor<1, dim> initial_direction_;
+  dealii::Tensor<1, 3>   initial_1d_state_;
+};
+
+
+template <int dim>
+class TimeStep : public dealii::ParameterAcceptor
+{
+public:
+  static constexpr unsigned int problem_dimension =
+    ProblemDescription<dim>::problem_dimension;
+
+  using rank1_type = typename ProblemDescription<dim>::rank1_type;
+  using rank2_type = typename ProblemDescription<dim>::rank2_type;
+
+  typedef std::array<dealii::LinearAlgebra::distributed::Vector<double>,
+                     problem_dimension>
+    vector_type;
+
+  TimeStep(const MPI_Comm &                   mpi_communicator,
+           dealii::TimerOutput &              computing_timer,
+           const grendel::OfflineData<dim> &  offline_data,
+           const grendel::InitialValues<dim> &initial_values,
+           const std::string &                subsection = "TimeStep");
+
+  void prepare();
+
+  double step(vector_type &U, double t);
+
+protected:
+  const MPI_Comm &     mpi_communicator_;
+  dealii::TimerOutput &computing_timer_;
+
+  dealii::SmartPointer<const grendel::OfflineData<dim>> offline_data_;
+  ACCESSOR_READ_ONLY(offline_data)
+
+  dealii::SmartPointer<const grendel::InitialValues<dim>> initial_values_;
+  ACCESSOR_READ_ONLY(initial_values)
+
+private:
+  dealii::SparseMatrix<double> dij_matrix_;
```

Comment:
Do we need to use the trailing underscores? We don't do that in any of the other codes...